code/modules/mob/living/basic/heretic/rust_spirit.dm

```diff
@@ -0,0 +1,81 @@
+/// Pretty simple mob which creates areas of rust and has a rust-creating projectile spell
+/mob/living/basic/heretic_summon/rust_spirit
+	name = "Rust Walker"
+	real_name = "Rusty"
+	desc = "A grinding, clanking construct which leaches life from its surroundings with every armoured step."
+	icon_state = "rust_walker_s"
+	icon_living = "rust_walker_s"
+	maxHealth = 75
+	health = 75
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_TURFS
+	speed = 1
+	ai_controller = /datum/ai_controller/basic_controller/rust_walker
+
+/mob/living/basic/heretic_summon/rust_spirit/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/footstep, FOOTSTEP_MOB_RUST)
+	var/datum/action/cooldown/spell/aoe/rust_conversion/small/conversion = new(src)
+	conversion.Grant(src)
+	ai_controller?.set_blackboard_key(BB_GENERIC_ACTION, conversion)
+
+	var/datum/action/cooldown/spell/basic_projectile/rust_wave/short/wave = new(src)
+	wave.Grant(src)
+	ai_controller?.set_blackboard_key(BB_TARGETTED_ACTION, wave)
+
+/mob/living/basic/heretic_summon/rust_spirit/setDir(newdir)
+	. = ..()
+	if(dir & NORTH)
+		icon_state = "rust_walker_n"
+	else if(dir & SOUTH)
+		icon_state = "rust_walker_s"
```

Comment:
Would be cool if this used `base_icon_state` / `update_icon_state` for vv memes.